src/clr-angular/forms/select/utils/select-dom-adapter.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable} from "@angular/core";
+
+// TODO: Shijir is moving the DOM Adapter outside of the Datagrid. While rebase, move the functions
+// from this class to there.
+
+@Injectable()
+export class SelectDomAdapter {
+    clearChildren(element: any): void {
+        while (element.firstChild) {
+            element.removeChild(element.firstChild);
```

Comment:
Why not just `element.innerHTML = ''`? That sounds way more efficient.

Reply:
I read here that `element.innerHTML = ''` is not as fast as we think it is: https://stackoverflow.com/a/3955238

Reply:
@youdz: not sure if you read the above link but I am assuming you have and there is no change required here.